kubernetes/troubleshooting/pulumi_workflow_pods_monitor.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+
+# Default values
+NAMESPACE="cmda"
+INTERVAL=1
+MAX_LOGS=10
+LOG_DIR="logs"
+
+# Parse command line arguments
+while getopts "n:i:m:d:" opt; do
+  case $opt in
+    n) NAMESPACE="$OPTARG" ;;
+    i) INTERVAL="$OPTARG" ;;
+    m) MAX_LOGS="$OPTARG" ;;
+    d) LOG_DIR="$OPTARG" ;;
+    \?) echo "Usage: $0 [-n namespace] [-i interval] [-m max_logs] [-d log_dir]" >&2; exit 1 ;;
+  esac
+done
+
+# Create log directory if it doesn't exist
+mkdir -p "$LOG_DIR"
+
+# Log rotation function
+rotate_logs() {
+    cd "$LOG_DIR" || exit
+    TOTAL_LOGS=$(ls pulumi_workflow_pods_monitor_*.log 2>/dev/null | wc -l)
+    while [ "$TOTAL_LOGS" -ge "$MAX_LOGS" ]; do
+        OLDEST_LOG=$(ls -t pulumi_workflow_pods_monitor_*.log | tail -1)
+        rm "$OLDEST_LOG"
+        TOTAL_LOGS=$((TOTAL_LOGS - 1))
+    done
+}
+
+# Ensure namespace is set
+if [ -z "$NAMESPACE" ]; then
+  echo "Error: Namespace must be set."
+  exit 1
+fi
+
+TIMESTAMP=$(date +%Y%m%d_%H%M%S)
+LOGFILE="$LOG_DIR/pulumi_workflow_pods_monitor_${TIMESTAMP}.log"
+
+echo "Starting Pulumi workflow pods monitoring. Log file: $LOGFILE"
+echo "Monitoring pods in namespace: $NAMESPACE"
+echo "Started at: $(date)" > "$LOGFILE"
+echo "Target namespace: $NAMESPACE" >> "$LOGFILE"
+echo "" >> "$LOGFILE"
+
+# Infinite loop to monitor the pod
+while true; do
+  rotate_logs
+  
+  PODS=$(kubectl get pods -n "$NAMESPACE" -l 'app.kubernetes.io/component=pulumi-workflow' --no-headers -o custom-columns=":metadata.name")
```

Comment:
Might be worth setting the label selector by variable in case this value changes.